### PR TITLE
Genewise

### PR DESCRIPTION
**Pull request by brentp**

### CODE REVIEW

File: docs/content/tools.rst [hunk 1]
```diff
@@ -659,6 +659,59 @@ are not):
...
+The gemini query tool allows querying by variant and the inheritance tools
+described above enable querying by gene for fixed inheritance patterns.
+The `gene_wise` tool allows querying by gene with custom genotype filters
+to bridge the gap between these tools.
+
+With this tool, multiple `--gt-filter` s can be specified. Each filter can
+be any valid filter; often, it will make sense to have 1 filter for each
+family. For example, given this pedigree:
+
+.. image : todo add image
```
Comment (bgruening) [inline docs/content/tools.rst]: I guess this should be fixed before merging.

File: gemini/genewise.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    all_cols = ["gts", "gt_types", "gt_phases", "gt_depths",
+                "gt_ref_depths", "gt_alt_depths", "gt_quals",
+                "gt_phred_ll_homref", "gt_phred_ll_het",
+                "gt_phred_ll_homalt"]
+    return [x for x in all_cols if x in gt_filter and not x in cols]
+
+
+def gen_results(rows, gt_filters, min_filters, min_variants, columns):
+    # we track the index of the passed filter in passed_filters.
+    gene_passed_filters = {}
```
Comment (bgruening) [inline gemini/genewise.py]: I think this can be a simple list, which holds the filter numbers.

File: gemini/genewise.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    header_printed = False
+    for groupkey, grp in it.groupby(gq, grouper):
+        grp = list(grp)
+        for x in gen_results(list(grp), cleaned_filters, min_filters or 0,
+                             min_variants, columns):
+            for c in added_cols:
+                if c != 'gene':
+                    del x.print_fields[c]
+            if not header_printed:
+                print "\t".join(x.print_fields.keys())
```
Comment (bgruening) [inline gemini/genewise.py]: debug lines?

Comment (brentp) [inline gemini/genewise.py]: these are printing the result of the evaluation

I'll add the image to the example that you pointed out above. Thanks!

File: gemini/genewise.py (hunk 2 above)
Comment (brentp) [inline gemini/genewise.py]: I'll leave it as-is for now as it won't be a bottle-neck and avoids issues with starting at 0.

**Pull request merged**